src/streaming/text/TextSourceBuffer.js

```diff
@@ -293,7 +293,8 @@ function TextSourceBuffer() {
 
         textTrackInfo.captionData = captionData;
         textTrackInfo.lang = mediaInfo.lang;
-        textTrackInfo.label = mediaInfo.id ? mediaInfo.id : mediaInfo.index; // AdaptationSet id (an unsigned int) as it's optionnal parameter, use mediaInfo.index
```

Comment:
Looks like you'll have to leave this line in place for 608 captions.

Reply:
Pardon me, since you've reassigned this as `.id`, the proper change would be in TextTracks.js to change https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/streaming/text/TextTracks.js#L501 to
```
if (textTrackQueue[i].id === trackId) {
```

Reply:
Well, I guess you have the choice of reverting to `.label` instead of `.id`, OR using `.id` and modifiying the following to use `.id` instead of `.label`:
https://github.com/Dash-Industry-Forum/dash.js/blob/e475fc0186399c3274006cb314ba6b8ef6cdcbfd/src/streaming/text/TextTracks.js#L98
https://github.com/Dash-Industry-Forum/dash.js/blob/e475fc0186399c3274006cb314ba6b8ef6cdcbfd/src/streaming/text/TextTracks.js#L491
https://github.com/Dash-Industry-Forum/dash.js/blob/e475fc0186399c3274006cb314ba6b8ef6cdcbfd/src/streaming/text/TextTracks.js#L501

Reply:
Updated @squarebracket